src/utils/proxy.ts

```diff
@@ -61,16 +61,22 @@ export async function proxyRequest(
     opts.headers,
   );
 
-  return sendProxy(event, target, {
-    ...opts,
-    fetchOptions: {
-      method,
-      body,
-      duplex,
-      ...opts.fetchOptions,
-      headers: fetchHeaders,
-    },
-  });
+  try {
+    return await sendProxy(event, target, {
+      ...opts,
+      fetchOptions: {
+        method,
+        body,
+        duplex,
+        ...opts.fetchOptions,
+        headers: fetchHeaders,
+      },
+    });
+  } catch {
+    event.node.res.statusCode = 502;
+    event.node.res.statusMessage = "Bad Gateway";
```

Comment:
I think only we can assume only proxy had been failed and perhaps we can double check if there is not already set statusCode and message on error (that currently is being ingored)

Additionally, this perhaps can be implemented directly in `sendProxy` util for consistent behavior

Reply:
Moving back to draft until implemented! Good idea 👍🏻